### PR TITLE
[search-react] versioned context

### PR DESCRIPTION
**Pull request by emmaindal**
Signed-off-by: Emma Indal <emma.indahl@gmail.com>

## Hey, I just made a Pull Request!

The context exported from the new search package (search-react) is now versioned using version bridge.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by iamEAP — COMMENTED** [review]
Whoops!  Thought I'd submitted feedback.  Just a couple of comments on changesets/docstrings.

File: plugins/search-react/src/context/SearchContext.tsx [hunk 1]
```diff
@@ -50,12 +53,29 @@ export type SearchContextState = {
...
+ */
+export const useSearch = () => {
+  const context = useContext(SearchContext);
+  if (!context) {
+    throw new Error('useSearch must be used within a SearchContextProvider');
+  }
+
+  const value = context.atVersion(1);
+  if (!value) {
+    throw new Error('No context for version 1 found');
```
Comment (Rugvip) [inline plugins/search-react/src/context/SearchContext.tsx:75]: Prolly mention `SearchContext v1` here instead, just to make it obvious what's going on if this error is encountered

File: plugins/search-react/src/context/SearchContext.tsx [hunk 2]
```diff
@@ -50,12 +53,29 @@ export type SearchContextState = {
...
 };
 
+// TODO(@backstage/techdocs-core): stop export this publicly when deprecating the context in @backstage/plugin-techdocs.
 /**
  * @public
  */
-export const SearchContext = createContext<SearchContextValue | undefined>(
-  undefined,
-);
+export const SearchContext = createVersionedContext<{
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/search-react/src/context/SearchContext.tsx:60]: Weirdly this isn't exported atm, so prolly drop the comment instead 😅

**Review by Rugvip — APPROVED** [review]
Looks good! 👍 

Just a minor nit on the error wording, and I think we can drop the TODO right?

File: plugins/search-react/src/context/SearchContext.tsx [hunk 3]
```diff
@@ -50,12 +53,29 @@ export type SearchContextState = {
   pageCursor?: string;
 };
 
+// TODO(@backstage/techdocs-core): stop export this publicly when deprecating the context in @backstage/plugin-techdocs.
```
[the remark this answers is not in the document]
Comment (emmaindal) [reply plugins/search-react/src/context/SearchContext.tsx:56]: Yeah haha now that you mention it I might go for that anyways. I started by doing that and then I was like, lets not release a new package, exporting this publicly and then mark as deprecated it in the next release, and thought it made more sense to actually deprecate it when we remove the context from `@backstage/plugin-techdocs` instead. But I'll do it now anyways!

**Comment by emmaindal** [comment]
@Rugvip ooooh I did not export it in the end, niiice! 😄

**Review by iamEAP — APPROVED** [review]
👏 👏 👏

**Pull request merged**